Update PULL_REQUEST_TEMPLATE.md

RussKie opened the pull request:
Include DCO ~and release under MIT~

Relates to #4737.

Further down the track we can consider enforcing sign-off with a [Probot](https://github.com/probot/dco).

amaiorano commented:
OMG sorry about the review approved spam! Last time I use my mobile browser to do this 😕

amaiorano commented:
> Are you SURE you approve? Just making sure. 😆

Apparently I've never approved of anything more in my life! :trollface:

RussKie commented:
@spdr870  @drewnoakes @sharwell waiting for your approvals

sharwell commented:
I'm neither for nor against this change. It would be good to update Contributing.md at the same time.

RussKie commented:
Slight update to include MIT licence and CONTRIBUTING.md

The PR template will have the following footer:
![image](https://user-images.githubusercontent.com/4403806/51518089-eba2db80-1e70-11e9-9698-2860a5982576.png)

CONTRIBUTING.md will have the following section added:
![image](https://user-images.githubusercontent.com/4403806/51518154-142ad580-1e71-11e9-8c36-3f0aff049c32.png)

I'm planning to merge this shortly.

vbjay commented:
Does this require sign offs?  If so, then I suggest we add the probot first.

RussKie commented:
No, no sign off for now. The presence of the above text should be sufficient.
No PRs without the above text should be merged in.

vbjay commented:
Also, should the license be replaced and not have dual licenses?

RussKie commented:
I'm not a lawyer, but I don't believe we are in a position to remove GPL licence, as the code is still licensed under it until we receive releases and waivers from all contributors past.

gerhardol commented:
It is possible to have dual licenses, but that should not be necessary here when switched.
However, in the meanwhile, contributors must accept to contribute under both MIT and GPL. I believe it is possible to re-licence MIT -> GPL (if MIT copyright notice is kept at least), so the wording should be OK.

All forks done before license is switched will be using GPL and it should be possible to import code that allows GPL (so code changes, maybe not external libraries).

drewnoakes commented:
Apologies all for missing this. @RussKie pinged me offline and I'm just seeing it now.

I don't want to block this, but does it make sense to include two incompatible licenses in the project? If I didn't know the history and stumbled upon them I might think it was dual licensed, which it isn't.

I approve of moving away from GPL and MIT seems fine to me. I wonder if getting a CLA in place makes more sense. I'm not a laywer obviously, but I believe a CLA could grant the maintainers the right to relicense contributions at some later point. That would allow changes from this point forward to be relicensed at the future point where we also have legal grounds to relicense historic commits.

sharwell commented:
@RussKie Easiest way to start with would be to use the approach of ANTLR:
https://github.com/antlr/antlr4/blob/master/contributors.txt

That would transfer forward to the CLA if the project joins the .NET Foundation.

RussKie commented:
Awesome, thanks Same

RussKie commented:
Take 3, I have updated the proposal following the ATLR's way.
I have taken my liberty and added all of you to the contributors.txt. 

Please let me know if you have any objections.

sharwell commented on contributors.txt:
```diff
@@ -0,0 +1,57 @@
+of any obligations to third parties and are granting Henk Westhuis and
+GitExtensions project contributors, henceforth referred to as The GitExtensions
+Project, a license to incorporate it into The GitExtensions Project 
+or related works under the BSD license. You understand that The GitExtensions
```
```suggestion
or related works under the MIT license. You understand that The GitExtensions
```

sharwell commented on contributors.txt:
```diff
@@ -0,0 +1,57 @@
+GitExtensions Project Contributors Certification of Origin and Rights
```
💭 The project isn't especially consistent on "GitExtensions" vs. "Git Extensions". I assume this is the preferred form?

RussKie replied:
Everywhere we use "Git Extensions", I'll fix this one up
Thank you

sharwell commented on contributors.txt:
```diff
@@ -0,0 +1,57 @@
+1. I am the creator of all my contributions. I am the author of all
+   contributed work submitted and further warrant and represent that
+   such work is my original creation and I have the right to license
+   it to The GitExtensions Project for release under the 3-clause BSD
```
```suggestion
   it to The GitExtensions Project for release under the MIT and/or GPL
```

This will simplify to only state MIT once we complete the history review.

sharwell commented on contributors.txt:
```diff
@@ -0,0 +1,57 @@
+2019/01/23, gerhardol, Gerhard Olsson, gerhard.nospam@gmail.com
+2019/01/23, drewnoakes, Drew Noakes, git@drewnoakes.com
+2019/01/23, amaiorano, Antonio Maiorano, amaiorano@gmail.com
+2019/01/23, sharwell, Sam Harwell, sam@tunnelvisionlabs.com
```
📝 We should probably all sign this individually

drewnoakes replied:
At the very least everyone should re-accept this PR post this change.

RussKie replied:
Yes, that was my plan.
I hope you're all ok with it.

vbjay commented:
@amaiorano  Only one approval this time?  :grinning:

amaiorano commented:
> @amaiorano Only one approval this time? 😀

Feeling less strongly about this one ;)

sharwell commented on contributors.txt:
```diff
@@ -0,0 +1,57 @@
+   license. I hereby grant The Git Extensions Project a nonexclusive,
+   irrevocable, royalty-free, worldwide license to reproduce,
+   distribute, prepare derivative works, and otherwise use this
+   contribution as part of the Git Extensions project, associated
```
💡 ... use ~~this contribution~~ **my contributions** as part ...

RussKie replied:
Thank you

RussKie commented:
To expedite the merger, I removed @jbialobr and @KindDragon from the list as I haven't received a re-approval from them. I will work with them to get their sign off.

vbjay commented:
What about submodules?

drewnoakes commented:
> What about submodules?

I guess that's less important as they're not GPL (afaik)...
